editor/gui/scene_tree_editor.cpp

```diff
@@ -1207,11 +1480,24 @@ Node *SceneTreeEditor::get_selected() {
 	return selected;
 }
 
-void SceneTreeEditor::set_marked(const HashSet<Node *> &p_marked, bool p_selectable, bool p_children_selectable) {
-	if (tree_dirty) {
-		_update_tree();
+void SceneTreeEditor::_update_marking_list(const HashSet<Node *> &p_marked) {
+	for (Node *N : p_marked) {
+		HashMap<Node *, CachedNode>::Iterator I = node_cache.get(N);
+		if (I) {
+			node_cache.mark_dirty(N);
+			node_cache.mark_children_dirty(N, true);
+		}
 	}
+}
+
+void SceneTreeEditor::set_marked(const HashSet<Node *> &p_marked, bool p_selectable, bool p_children_selectable) {
+	_update_if_clean();
+
+	_update_marking_list(marked);
```

Comment:
I am getting a crash within this function occasionally when opening the signal connection dialog due to garbage data in `marked`. Is calling `_update_marking_list()` twice intended? Commenting out the first call seems to fix the issue, but I can't tell if that breaks anything.

Reply:
Yes, calling it twice is intentional. Do you have a back trace of the crash?

Reply:
There should never be "garbage data" in `marked` that is a pre-existing bug then, but I can't figure out how garbage data could get into it.

Reply:
@hilloftheking can you try adding `marked.clear()` to https://github.com/godotengine/godot/blob/master/editor/gui/scene_tree_editor.cpp#L909 and see if this solves the issue?

Reply:
Sorry, I'm away for a few days. I should have opened an issue for this but I couldn't make a decent reproduction. I think it has to do with the connection dialog keeping data from a closed scene if anyone else can make a reproduction project. If not, I will make one when I get back.

Reply:
It does keep data from a closed scene until it reopens, this is by design because updating the invisible SceneTreeEditors is very costly. It makes everything take 6x as long! :)

I will see if I can come up with a fix for what I hypothesize is the problem.